pipeline/authn/authenticator_cookie_session.go

```diff
@@ -171,13 +175,14 @@ func forwardRequestToSessionStore(r *http.Request, checkSessionURL string, prese
 		return nil, helper.ErrForbidden.WithReason(err.Error()).WithTrace(err)
 	}
 
+	body, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return json.RawMessage{}, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("Unable to read session context from remote: %+v", err))
+	}
+
 	if res.StatusCode == 200 {
-		body, err := ioutil.ReadAll(res.Body)
-		if err != nil {
-			return json.RawMessage{}, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("Unable to fetch cookie session context from remote: %+v", err))
-		}
 		return body, nil
-	} else {
-		return json.RawMessage{}, errors.WithStack(helper.ErrUnauthorized)
 	}
+
+	return json.RawMessage{}, errors.WithStack(helper.ErrUnauthorized.WithReasonf("API said: %d -- %+v", res.StatusCode, body))
```

Comment:
Please return a generic error here. Printing out the API response to the caller is dangerous, as it can be used to discover the API responses of the API endpoint by someone who is not allowed to see them usually.